kakao/src/main/kotlin/com/agoda/kakao/spinner/KSpinner.kt

```diff
@@ -0,0 +1,246 @@
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     *
+     * @see ViewBuilder
+     */
+    constructor(builder: ViewBuilder.() -> Unit, itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit) {
+        val vb = ViewBuilder().apply(builder)
+        matcher = vb.getViewMatcher()
+        view = vb.getViewInteractionDelegate()
+        itemTypes = KAdapterItemTypeBuilder().apply(itemTypeBuilder).itemTypes
+    }
+
+    /**
+     * Constructs view class with parent and view interaction from given ViewBuilder
+     *
+     * @param parent Matcher that will be used as parent in isDescendantOfA() matcher
+     * @param builder ViewBuilder which will result in view's interaction
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     *
+     * @see ViewBuilder
+     */
+    constructor(
+        parent: Matcher<View>, builder: ViewBuilder.() -> Unit,
+        itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit
+    ) : this({
+                 isDescendantOfA { withMatcher(parent) }
+                 builder(this)
+             }, itemTypeBuilder)
+
+    /**
+     * Constructs view class with parent and view interaction from given ViewBuilder
+     *
+     * @param parent DataInteraction that will be used as parent to ViewBuilder
+     * @param builder ViewBuilder which will result in view's interaction
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     *
+     * @see ViewBuilder
+     */
+    @Suppress("UNCHECKED_CAST")
+    constructor(
+        parent: DataInteraction, builder: ViewBuilder.() -> Unit,
+        itemTypeBuilder: KAdapterItemTypeBuilder.() -> Unit
+    ) {
+        val makeTargetMatcher = DataInteraction::class.java.getDeclaredMethod("makeTargetMatcher")
+        val parentMatcher = makeTargetMatcher.invoke(parent)
+
+        val vb = ViewBuilder().apply {
+            isDescendantOfA { withMatcher(parentMatcher as Matcher<View>) }
+            builder(this)
+        }
+
+        matcher = vb.getViewMatcher()
+        view = vb.getViewInteractionDelegate()
+        itemTypes = KAdapterItemTypeBuilder().apply(itemTypeBuilder).itemTypes
+    }
+
+    /**
+     * Performs given actions/assertion on child at given position
+     *
+     * @param T Type of item at given position. Must be registered via constructor.
+     * @param position Position of item in adapter
+     * @param function Tail lambda which receiver will be matched item with given type T
+     */
+    inline fun <reified T : KAdapterItem<*>> childAt(position: Int, function: T.() -> Unit) {
+        val provideItem = itemTypes.getOrElse(T::class) {
+            throw IllegalStateException("${T::class.java.simpleName} did not register to Spinner")
+        }.provideItem
+
+        val interaction = Espresso.onData(Matchers.anything())
+            .inRoot(isPlatformPopup())
+//            .inAdapterView(matcher)
```

Comment:
It seems this code is unnecessary. Although, to reduce the chance of multiple matching exception, you can bind this interaction to the root view class of the popup as well.

Reply:
I haven't understood this. Isn't `isPlatformPopup` already doing the binding?

Reply:
That is the root matcher, basically it matches the windows inside the windows stack of your app. But you might have more than 1 windows opened alongside your main content's window at the same time.
In that situation you will get multiple matching exception.
So, if it's possible to distinctively detect the root view of a popup window, we should do that.

Reply:
I see. The only way I can think of distinctively identifying popup root view is checking if's a descendant of of the DropDownView. Do you have a better constraint I should implement?

Reply:
I think that should do cover most of the cases.

Reply:
@Unlimity I've added the bounds.